File: 1748-sum-of-unique-elements/1748-sum-of-unique-elements.java
class Solution {
    public int sumOfUnique(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int sum = 0;

        for(int i = 0; i < nums.length; i++){
            if(!map.containsKey(nums[i]))
                map.put(nums[i], 1);
            else
                map.put(nums[i], map.get(nums[i]) + 1);
        }

        Integer[] arr = map.keySet().toArray(new Integer[0]);
        for(int i = 0; i < map.size(); i++){
            if(map.get(arr[i]) == 1)
                sum += arr[i];
        }
        return sum;
    }
}

File: 1748-sum-of-unique-elements/1748-sum-of-unique-elements.swift
class Solution {
    func sumOfUnique(_ nums: [Int]) -> Int {
        var arr = nums
        var res = 0
        var flag = 0
        arr.sort()
        
        for (i,val) in arr.enumerated() {
            if (flag == 0 && i != arr.count-1 && arr[i+1] != arr[i]) {
                res += arr[i]
            } else if i != arr.count-1 && arr[i] == arr[i+1]{
                flag = 1
            } else if i == arr.count-1 && flag == 0 {
                res += arr[i]
            } else {
                flag = 0
            }
        }
        return res
    }
}